devlink_linux.go

```diff
@@ -581,6 +583,11 @@ func (port *DevlinkPort) parseAttributes(attrs []syscall.NetlinkRouteAttr) error
 						port.Fn = &DevlinkPortFn{}
 					}
 					port.Fn.OpState = uint8(nested.Value[0])
+				case MLXDEVM_PORT_FN_ATTR_TRUST:
+					if port.Fn == nil {
+						port.Fn = &DevlinkPortFn{}
+					}
+					port.Fn.Trust = uint8(nested.Value[0])
```

Comment:
we should not parse MLXDEVM_PORT_FN_ATTR_TRUST in case the socket provided is devlink

perhaps this function should have `extraPortFunctionAttributes` as an argument which will contain `MLXDEVM_PORT_FN_ATTR_TRUST` enum value in case mlxdevm gen-netlink family is used.

Reply:
fixed